### PR TITLE
Remove patch from pybind11

### PR DESCRIPTION
**Pull request by dagardner-nv**
## Description
* Removes pybind11 patch
* All MRC & Morpheus tests are passing with this change

Part of nv-morpheus/morpheus#961

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by mdemoret-nv — CHANGES_REQUESTED** [review]
Have you tested using pybind11 from Conda?

**Comment by dagardner-nv** [comment]
> Have you tested using pybind11 from Conda?

Yes I did, I tested two setups:
1) As-is without the patch installed via CPM
2) Using conda installed pybind11 which was the newish v2.10.4

Both worked, but I wanted to do it as a two-step

File: cmake/morpheus_utils/package_config/pybind11/Configure_pybind11.cmake [hunk 1]
```diff
@@ -26,18 +26,17 @@ function(morpheus_utils_configure_pybind11)
...
-      GIT_TAG         "v${PYBIND11_VERSION}"
-      GIT_SHALLOW     TRUE
-      PATCH_COMMAND   git checkout -- . && git apply --whitespace=fix ${CMAKE_CURRENT_FUNCTION_LIST_DIR}/patches/enable_fiber_support.patch
-      OPTIONS         "PYBIND11_INSTALL ON"
-                      "PYBIND11_TEST OFF"
+    GIT_REPOSITORY https://github.com/pybind/pybind11.git
+    GIT_TAG "v${PYBIND11_VERSION}"
+    GIT_SHALLOW TRUE
+    OPTIONS "PYBIND11_INSTALL ON"
+    "PYBIND11_TEST OFF"
```
Comment (mdemoret-nv) [inline cmake/morpheus_utils/package_config/pybind11/Configure_pybind11.cmake:39]: Keep these indentations where they were before

**Review by mdemoret-nv — APPROVED** [review]

**Pull request merged**